### DISCUSSION TITLE
Any way to access the `PyTypeObject` created by a `class_`?

### DISCUSSION THREAD

**Discussion by gmgunter** [pybind/pybind11#3323] [Q&A] [answered]
Is there any way to get a pointer to the underlying `PyTypeObject` that's created when a new `class_` is registered? 

This would be useful, for example, if a user wanted to register a class as a new numpy dtype, which requires creating and registering a [`PyArray_Descr`](https://numpy.org/doc/stable/reference/c-api/types-and-structures.html#c.PyArray_Descr) object that stores a pointer to the type object.

I looked through the implementation of [`generic_type::initialize()`](https://github.com/pybind/pybind11/blob/b3573ac9615b705d6e5b9bf598a3378edf079f06/include/pybind11/pybind11.h#L1132), where the new type appears to be created, and it doesn't seem to store or return a pointer to the type object in a way that it could be accessed later, but perhaps I missed something. If not, would it be possible to expose this at some point in the future?

**Comment by Skylion007** [comment, marked as answer]
Off the top of my head, you may be able to use PyBind11 type object through the `py::type::of<MyClass>()` feature: https://pybind11.readthedocs.io/en/stable/advanced/classes.html#accessing-the-type-object and then access the underlying PyTypeObject by calling `.ptr()` or `.release()`.

**Comment by gmgunter** [comment]
Thanks! That seems to be exactly what I was looking for
